README.md

```diff
@@ -37,10 +37,302 @@ If you've used Elasticsearch with gems like [Searchkick](https://github.com/anka
+  TRANSACTION (0.5ms)  commit transaction                                                                                 
+=> #<User id: 1, created_at: "2022-05-26 05:29:31.894710000 +0000", updated_at: "2022-05-26 05:29:31.894710000 +0000", email: [FILTERED], name: [FILTERED], dob: [FILTERED]>      
+```
+
+Although our stored data is secure, we are not able to query it.
+
+```bash
+irb(main):003:0> User.where(email: "grace@hopper.example").count
+  User Count (0.4ms)  SELECT COUNT(*) FROM "users" WHERE "users"."email" = ?  [["email", "grace@hopper.example"]]
+=> 0
+```
+
+This is what ActiveStash solves.
+
+Every time you create or update a record, the data is indexed to CipherStash via ActiveRecord callbacks.
+
+Queries are delegated to CipherStash but return ActiveRecord models so things just work.
+
+Go to [getting started with ActiveStash](#getting-started-with-activestash) to learn how.
+
+
+## Getting started with ActiveStash
+
+1. Add active stash to your gem file
+
+```
+gem "active_stash"
+```
+
+2. Run bundle install
+
+```
+➜  bundle install
+```
+
+3. Create a CipherStash account, get a workspace and login using the below rake task.
+
+https://docs.cipherstash.com/tutorials/getting-started/create-an-account.html
+
+```bash
+➜  rake active_stash:login[YOURWORKSPACEID]
+```
+
+Note: If you are using `zsh` you may need to escape the brackets
+
+```bash
+rake active_stash:login\['WorkspaceId'\] 
+```
+
+4. Any model in which you include ActiveStash::Search, will need to have a `stash_id` column added of type `string`.
+
+To add this to the table underlying your `User` model add the below migration:
+
+```sh
+$ rails g migration AddStashIdToUser stash_id:string:index
+$ rails db:migrate
+```
+
+The above command also ensures that an index is created on `stash_id`.
+
+
+5. Update your user model to include the ActiveStash search module and declare which fields are searchable.
+
+
+```ruby
+# app/models/user.rb
+class User < ApplicationRecord
+  include ActiveStash::Search
+
+  encrypts :name, :email
+  encrypts :dob, type: :date
+
+  # Fields that will be indexed into CipherStash
+  stash_index :name, :email, :dob
+
+  self.ignored_columns = %w[name email dob]
+end
+```
+
+6. Reindex your existing data into CipherStash with ActiveStash
+
+```
+➜  rails active_stash:reindexall
+```
+
+7. Try querying the same user we created earlier
+
+```bash
+$ rails c
+ >> User.query(email: "grace@example.com").count
+ => 1
```

Comment:
```suggestion
$ rails c 
 >> User.where(email: "grace@example.com").count
 => 0 # no records, because the database isn't searchable
 >> User.query(email: "grace@example.com").count
 => 1 # a record, because CipherStash makes your encrypted database searchable$ rails c
```